Fix issue2534

gregueiras opened the pull request:
<!--
Before submitting this PR, please make sure that:
- You have read and understand the contributing.md
- You have checked docs/code_style.md for information on code style
-->
## Description
Users can now schedule a theme to be applied
![image](https://user-images.githubusercontent.com/28135630/49227217-7a026000-f3e0-11e8-8558-0423d2c03b99.png)
In this example, from 07:20 to 17:50 the theme to be used is _Monokai_, otherwise the Default theme is used
<!--
Tell us what your PR does.
Please attach a screenshot/ video/gif image describing your PR if possible.
-->

## Issue fixed
<!--
Please list out all issue fixed with this PR here.
-->
Issue #2534 Scheduled themes or auto day/night mode
<!--
Please make sure you fill in these checkboxes,
your PR will be reviewed faster if we know exactly what it does.

Change :white_circle: to :radio_button: in all the options that apply
-->
## Type of changes

- :radio_button: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :white_circle: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

Note: I couldn't run the tests

gregueiras commented:
@ZeroX-DG Thank you 😄

ZeroX-DG commented:
@gregueiras can you resolve all the conflicts?

gregueiras commented:
> @gregueiras can you resolve all the conflicts?

I've resolved all the conflicts and changed the scheduled theme dropdown options to be the same as the default theme one.

ZeroX-DG commented on browser/main/modals/PreferencesModal/UiTab.js:
```diff
@@ -207,6 +215,43 @@ class UiTab extends React.Component {
+    if (rangeSlider && rangeBullet) {
+      const bulletPosition = rangeSlider.value / rangeSlider.max
+      rangeBullet.style.left =
+        bulletPosition * 574 + 6 * (1 - bulletPosition) + 'px'
```
Can you explain to me where the value `574` come from?

gregueiras replied:
We thought that we needed to set the slider position, but I tried removing it and it is working as expected, so I've removed the handleSlider function and we can just use the handleUIChange

gregueiras commented on browser/main/modals/PreferencesModal/UiTab.js:
```diff
@@ -207,6 +215,43 @@ class UiTab extends React.Component {
   }
 
+  /**
+   * Gets the total number of minutes and returns a string in the HH:MM format
```
I've removed that comment

gregueiras commented:
I've fixed the problem. It was just the way I was comparing the dates. If you scheduled the change to happen at 15:30, it would change only at 15:31.

gregueiras commented on browser/main/lib/ThemeManager.js:
```diff
@@ -10,8 +10,8 @@ const chooseTheme = ui => {
-  const isBetweenStartAndEnd = minutes >= start && minutes <= end
-  const isBetweenEndAndStart = minutes >= start || minutes <= end
+  const isBetweenStartAndEnd = minutes >= start && minutes < end
+  const isBetweenEndAndStart = minutes >= start || minutes < end
```
I've fixed the problem. It was just the way I was comparing the dates. If you scheduled the change to happen at 15:30, it would change only at 15:31.

ZeroX-DG commented:
@gregueiras it's still not working for me. I'm not quite sure what is causing it because at first the `setInterval` doesn't work so I removed `this.refreshTheme=` and it start running but it still doesn't change the theme. I was on setting page the theme didn't change, not sure if it change on main page.

gregueiras commented:
Hey, sorry about the delay.

I couldn't reproduce your problem... I thought it could be some bug regarding timezone, but even after changing it, it's still working

![Peek 2020-03-24 10-20](https://user-images.githubusercontent.com/28135630/77384863-0e07f000-6db9-11ea-9c57-d71520c623dc.gif)


Do you any other idea?

ZeroX-DG commented:
@gregueiras The problem that I'm having is the incorrect theme from config. I added this line in your refresh theme loop.
![image](https://user-images.githubusercontent.com/12984316/77702752-16149980-701e-11ea-91e9-79807b79fdf0.png)
And this is what been logging to the terminal
![Peek 2020-03-27 11-25](https://user-images.githubusercontent.com/12984316/77702782-24fb4c00-701e-11ea-8cd0-389934dd00d9.gif)
Do you have any idea for what is causing this issue?

gregueiras commented:
I believe I've fixed it!
![Screenshot from 2020-03-27 00-01-01](https://user-images.githubusercontent.com/28135630/77708008-c05cd480-6fbe-11ea-9db9-7d33195845f2.png)


Previously, I was changing the instance returned from ConfigManager, but wasn't really setting it anywhere. I've changed ThemeManager to return the new configuration if it changes, and then we can set the new config on the loop. I've tought about setting it directly on ThemeManager, but as ConfigManager imports it, it would create a circular dependency. 

What do you think?
